proposals/0219-ios-check-type.md

```diff
@@ -30,18 +30,33 @@ For example:
 
 ## Proposed solution
 
-Remove method :
+Mark method as private:
 `-(nullable id)sdl_objectForName:(SDLName)name;`. 
 
-Instead of the removed method, use this existing method:
+Instead of method above use next methods:
+
+For optional values, can be used functions that return object of classType or nil:
 `- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType;`.
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType`.
+
+For mandatory values, next functions try to return object of classType overwise error would be filled. 
+In case, when error isn't `nil`, returned object can be not type of `classType`:
+`- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error`
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error;`.
 
 For suitable working with SDLEnums create the following methods:
-`- (nullable SDLEnum)sdl_enumForName:(SDLName)name`
+`- (nullable SDLEnum)sdl_enumForName:(SDLName)name;`
+`- (nullable SDLEnum)sdl_enumForName:(SDLName)name error:(NSError *);`.
 `- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name;`
```

Comment:
This can be removed, per the requested revisions.

Reply:
This method calls inside method with error, like example above